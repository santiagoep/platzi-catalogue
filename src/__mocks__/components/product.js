export default {
  slug: 'slug',
  name: 'name',
  price: 124,
  image: { src: '/icons/isotype.png', alt: 'Isotype' },
  hoverImage: { src: '/icons/isotype.png', alt: 'Isotype' }
};
