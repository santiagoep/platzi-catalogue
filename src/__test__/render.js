/* istanbul ignore file */
/* eslint-disable import/no-extraneous-dependencies */
import { Provider } from 'react-redux';
import { mount, shallow } from 'enzyme';
import Renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import { makeMockedStore } from '@store/index';
import mockedStore from '@__mocks__/store/index';
import lightTheme from '@assets/styles/themes/light';

const WrapperComponent = ({ children, config, initialState }) => {
  if (config && config.theme && config.redux) {
    const store = makeMockedStore(initialState || mockedStore);
    return (
      <Provider store={store}>
        <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>
      </Provider>
    );
  }
  if (config && config.redux) {
    const store = makeMockedStore(initialState || mockedStore);
    return <Provider store={store}>{children}</Provider>;
  }
  if (config && config.theme) {
    return <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>;
  }
  return children;
};

const renderWithProviders = (
  { redux, theme } = {},
  initialState,
  getMockedStore
) => {
  const renderProviders = (Component) => (
    <WrapperComponent
      config={{ redux, theme }}
      initialState={initialState}
      getMockedStore={getMockedStore}
    >
      {Component}
    </WrapperComponent>
  );

  return {
    mount: (Component) => mount(renderProviders(Component)),
    render: (Component) => render(renderProviders(Component)),
    shallow: (Component) => shallow(renderProviders(Component)),
    renderer: (Component) => Renderer.create(renderProviders(Component))
  };
};

export default renderWithProviders;
