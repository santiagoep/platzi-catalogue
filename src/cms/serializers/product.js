import imagesSerializer from './common/images';
import validateField from '@utils/validations';

const productsSerializer = ({ product: { images, description, ...rest } }) =>
  validateField(
    {
      ...rest,
      description: validateField(description?.text, ''),
      images: imagesSerializer({ images })
    },
    {}
  );

export default productsSerializer;
