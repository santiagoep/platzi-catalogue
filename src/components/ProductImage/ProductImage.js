import Image from 'next/image';
import PropTypes from 'prop-types';

const ProductImage = ({ src, alt, ...rest }) => (
  <Image
    quality={100}
    loading='lazy'
    layout='fill'
    objectFit='cover'
    src={src}
    alt={alt}
    {...rest}
  />
);

ProductImage.propTypes = {
  src: PropTypes.string.isRequired,
  alt: PropTypes.string
};

ProductImage.defaultProps = {
  alt: ''
};

export default ProductImage;
