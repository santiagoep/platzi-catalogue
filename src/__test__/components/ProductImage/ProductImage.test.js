import render from '../../render';
import product from '@__mocks__/components/product';
import ProductImage from '@components/ProductImage/ProductImage';

describe('<ProductImage />', () => {
  it('Should render snapshot', () => {
    const { mount } = render({ theme: true });
    const component = mount(<ProductImage {...product.image} />);
    expect(component).toMatchSnapshot();
  });
});
