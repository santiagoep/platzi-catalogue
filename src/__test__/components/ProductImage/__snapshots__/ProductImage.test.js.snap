// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductImage /> Should render snapshot 1`] = `
<WrapperComponent
  config={
    Object {
      "redux": undefined,
      "theme": true,
    }
  }
>
  <Component
    theme={
      Object {
        "borderRadius": Object {
          "button": "100px",
          "large": "18px",
          "medium": "8px",
          "none": "unset",
          "small": "3px",
          "total": "50%",
        },
        "boxShadows": Object {
          "main": "1px 2px 4px 0 rgba(21,99,157,.16)",
          "secondary": "0 0 0 1px #0791e6",
        },
        "breakpoints": Object {
          "desktop": "@media only screen and (min-width: 1280px)",
          "laptop": "@media only screen and (min-width: 1024px)",
          "mobile": "@media only screen and (min-width: 320px)",
          "screen": "@media only screen and (min-width: 1920px)",
          "smallMobile": "@media only screen and (min-width: 0px)",
          "tablet": "@media only screen and (min-width: 640px)",
        },
        "colors": Object {
          "bg": Object {
            "primary": "linear-gradient(90deg,#1c3643,#273b47 25%,#1e5372)",
            "quaternary": "#F3F8FB",
            "secondary": "white",
            "tertiary": "#0791e6",
          },
          "border": Object {
            "primary": "#A8B1C0",
            "secondary": "white",
          },
          "font": Object {
            "primary": "#131D30",
            "quaternary": "#A8B1C0",
            "secondary": "#98ca3f",
            "tertiary": "white",
          },
          "separator": Object {
            "primary": "#A8B1C0",
            "secondary": "#0791e6",
          },
        },
        "containerWidth": Object {
          "desktop": Object {
            "maxWidth": "85%",
            "width": "85%",
          },
          "fluid": Object {
            "maxWidth": "100%",
            "width": "100%",
          },
          "limitFluid": Object {
            "maxWidth": "1920px",
            "width": "100%",
          },
          "mobile": Object {
            "maxWidth": "92%",
            "width": "92%",
          },
          "screen": Object {
            "maxWidth": "75%",
            "width": "75%",
          },
          "smallMobile": Object {
            "maxWidth": "92%",
            "width": "92%",
          },
          "tablet": Object {
            "maxWidth": "85%",
            "width": "85%",
          },
        },
        "font": Object {
          "family": Object {
            "primary": "'Lato', sans-serif",
          },
          "letterSpacing": Object {
            "medium": "0.5rem",
            "normal": "0rem",
          },
          "lineHeight": Object {
            "medium": "0.6rem",
            "normal": "normal",
          },
          "size": Object {
            "h1": "3.75rem",
            "h2": "3rem",
            "h3": "2.5rem",
            "h4": "2.25rem",
            "h5": "1.625rem",
            "h6": "1.5rem",
            "help": "0.625em",
            "paragraph1": "1.125rem",
            "paragraph2": "1rem",
            "paragraph3": "0.875rem",
            "paragraph4": "0.75rem",
            "small": "0.5rem",
            "subtitle1": "1.1375rem",
            "subtitle2": "1.25rem",
            "subtitle3": "1.375rem",
          },
        },
        "name": "LIGHT",
        "transitions": Object {
          "button": "background-color 0.4s ease, color 0.4s ease, border-color 0.4s ease",
        },
        "utils": Object {
          "spacing": [Function],
        },
        "zIndex": Object {
          "backward": -1,
          "fixed": 2,
          "forward": 1,
          "higher": 3,
          "initial": 0,
          "super": 100,
        },
      }
    }
  >
    <ProductImage
      alt="Isotype"
      src="/icons/isotype.png"
    >
      <Image
        alt="Isotype"
        layout="fill"
        loading="lazy"
        objectFit="cover"
        quality={100}
        src="/icons/isotype.png"
      >
        <div
          style={
            Object {
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "left": 0,
              "margin": 0,
              "overflow": "hidden",
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <img
            alt="Isotype"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": "cover",
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "visibility": "hidden",
                "width": 0,
              }
            }
          />
        </div>
      </Image>
    </ProductImage>
  </Component>
</WrapperComponent>
`;
